dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java

```diff
@@ -334,19 +367,24 @@ private OutputStream compress( HttpServletResponse response, String attachment,
 
         if ( Compression.GZIP == compression )
         {
+            fileName = fileName.replace( "." + format + ".gzip", "" );
+            fileName = fileName.replace( "." + format + ".gz", "" );
+
             response.setHeader( ContextUtils.HEADER_CONTENT_DISPOSITION,
-                "attachment; filename=" + fileName + "." + format + ".gzip" );
+                "attachment; filename=" + fileName + "." + format + ".gz" );
             response.setHeader( ContextUtils.HEADER_CONTENT_TRANSFER_ENCODING, "binary" );
             return new GZIPOutputStream( response.getOutputStream() );
         }
         else if ( Compression.ZIP == compression )
         {
+            fileName = fileName.replace( "." + format + ".zip", "" );
```

Comment:
Cannot quite follow the idea here. If the requested file already has a e.g. `xyz.json.zip` in the name we strip that part?
Then later we use `fileName` again and append `.json` so we send the file originally named `xyz.json.zip` as `xyz.json`? Something does not match up for me here. But that might be me not getting the intention here.

Reply:
The client shouldn't really be sending in `.xml.zip` at all, but our current import/export app does exactly that. So it seemed cleaner to make sure we don't have any additional suffix (except for wanted filename) and then add it back where it makes sense. We could of course check for extension etc, but this seems like a simpler solution.

A more appropriate use of this endpoint is to send `attachment=fileName&compression=zip&format=json` and then we take care of the rest (including naming it appropriately in the zip file).